azurerm/resource_arm_notification_hub_namespace.go

```diff
@@ -224,16 +260,37 @@ func resourceArmNotificationHubNamespaceDelete(d *schema.ResourceData, meta inte
 	return nil
 }
 
-func expandNotificationHubNamespacesSku(input []interface{}) *notificationhubs.Sku {
-	v := input[0].(map[string]interface{})
+func expandNotificationHubNamespacesSku(d *schema.ResourceData) *notificationhubs.Sku {
+	v := d.Get("sku_name").(string)
+
+	// Remove in 2.0
+	if v == "" {
+		inputs := d.Get("sku").([]interface{})
+
+		if len(inputs) == 0 {
+			return nil
+		}
 
-	skuName := v["name"].(string)
+		input := inputs[0].(map[string]interface{})
+		v = input["name"].(string)
+	}
 
-	return &notificationhubs.Sku{
-		Name: notificationhubs.SkuName(skuName),
+	sku := notificationhubs.Sku{
+		Name: notificationhubs.SkuName(v),
 	}
+
+	return &sku
+}
+
+func flattenNotificationHubNamespacesSkuName(sku *notificationhubs.Sku) string {
+	if sku == nil {
+		return ""
+	}
+
+	return string(sku.Name)
 }
 
+// Remove in 2.0
 func flattenNotificationHubNamespacesSku(input *notificationhubs.Sku) []interface{} {
```

Comment:
As can this

Reply:
Keeping this one, as I think it is better to set `sku` and `sku_name` independent of each other...